File: app/src/main/java/com/spiderbiggen/randomchampionselector/storage/database/converters/StringArrayConverter.java
package com.spiderbiggen.randomchampionselector.storage.database.converters;

import android.arch.persistence.room.TypeConverter;
import android.text.TextUtils;

/**
 * Created on 1-3-2018.
 *
 * @author Stefan Breetveld
 */

public class StringArrayConverter {

    private static final String DELIMITER = ",";

    @TypeConverter
    public static String[] toStringArray(String value) {
        return value == null ? null : value.split(DELIMITER);
    }

    @TypeConverter
    public static String toString(String[] values) {
        return values == null ? null : TextUtils.join(DELIMITER, values);
    }
}

File: app/src/main/java/com/spiderbiggen/randomchampionselector/data/storage/database/converters/StringArrayConverter.kt
package com.spiderbiggen.randomchampionselector.data.storage.database.converters

import android.arch.persistence.room.TypeConverter
import android.text.TextUtils

/**
 * Created on 1-3-2018.
 *
 * @author Stefan Breetveld
 */

class StringArrayConverter {

    @TypeConverter
    fun toStringArray(value: String?): Array<String>? {
        return value?.split(DELIMITER)?.dropLastWhile { it.isEmpty() }?.toTypedArray()
    }

    @TypeConverter
    fun toString(values: Array<String>?): String? {
        return if (values == null) null else TextUtils.join(DELIMITER, values)
    }

    companion object {
        private const val DELIMITER = ","
    }
}
